mitosheet/src/components/modals/ReplayAnalysisModals.tsx

```diff
@@ -0,0 +1,145 @@
+                                    ...prevUIState,
+                                    currOpenModal: {type: ModalEnum.None}
+                                }
+                            })
+                            // TODO: do we want to change the parameter here, so they don't get overwritten?
+                            return true;
+                        }}
+                    >
+                        Get Immediate Support
+                    </TextButton>
+                </> 
+            }
+        />
+    )    
+};
+
+/*
+    This modal displays to the user when the analysis that they are 
+    replaying fails to run correctly.
+*/
+export const InvalidReplayedAnalysisModal = (
+    props: {
+        error: MitoError | undefined, 
+        setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+        mitoAPI: MitoAPI;
+        oldAnalysisName: string;
+        newAnalysisName: string;
+    }): JSX.Element => {
+    const [viewTraceback, setViewTraceback] = useState(false);
+
+    if (props.error === undefined) {
+        return (<></>)
+    }
+
+    return (
+        <DefaultModal
+            header={"Analyis Could Not Be Replayed"}
+            modalType={ModalEnum.Error}
+            wide
+            viewComponent={
+                <Fragment>
+                    {props.error.to_fix &&
+                        <div className='text-align-left text-body-1' onClick={() => setViewTraceback((viewTraceback) => !viewTraceback)}>
+                            There was an error in replaying your analysis. The generated code can be found below, but this anlaysis could not be replayed on the new dataset. {' '}
+                            {props.error.traceback && 
+                                <span className='text-body-1-link'>
+                                    Click to view full traceback.
+                                </span>
+                            }
+                        </div>
+                    }
+                    {props.error.traceback && viewTraceback &&
+                        <div className='flex flex-column text-align-left text-overflow-hidden text-overflow-scroll mt-5px' style={{height: '200px', border: '1px solid var(--mito-purple)', borderRadius: '2px', padding: '5px'}}>
+                            <pre>{props.error.traceback}</pre>
+                        </div>
+                    }
+                </Fragment>
+            }
+            buttons={
+                <>
+                    <TextButton
+                        variant='dark'
+                        width='medium'
+                        onClick={() => {
+                            /**
+                             * We also need to actually change the analysis to replay in the code cell, 
+                             * so that we know something happened to invalidate this analysis to replay,
+                             * and we can write new code?
+                             */
```

Comment:
Changing the analysisName wouldn't get us anything here - it only changes if you replay an analysis successfully currently. 

The analysis name in the mitosheet.sheet call (and being used in the generated code) needs to be different if there is an invalid/nonexistant analysis passed, as otherwise we overwrite that code (which we don't want to do). 

Since the analysisName is different than the analysis_to_replay param in the case of failure/non-existance, we use the analysisName to be this new thing!